core/src/main/java/nu/yona/server/goals/service/GoalDTO.java

```diff
@@ -56,6 +54,11 @@ public boolean isMandatory()
 
 	public abstract Goal createGoalEntity();
 
+	protected static UUID determineActivityCategoryID(String activityCategoryUrl)
+	{
+		return UUID.fromString(activityCategoryUrl.substring(activityCategoryUrl.lastIndexOf('/') + 1));
```

Comment:
But after deserialization, you can also parse the `activityCategoryUrl` at the controller and after that set an `activityCategoryID` on the same DTO.